src/engraving/dom/fret.cpp

```diff
@@ -281,14 +281,23 @@ void FretDiagram::setStrings(int n)
     m_markers = tempMarkers;
 
     for (int fret = 1; fret <= m_frets; ++fret) {
-        if (barre(fret).exists()) {
-            if (m_barres[fret].startString + difference <= 0) {
-                removeBarre(fret);
-                continue;
+        if (!barre(fret).empty()) {
+            
+            for (auto& b : m_barres[fret]) {
+                if (b.startString + difference <= 0) {
+                    removeBarre(fret);
+                    break;
+                }
             }
-
-            m_barres[fret].startString = std::max(0, m_barres[fret].startString + difference);
-            m_barres[fret].endString   = m_barres[fret].endString == -1 ? -1 : m_barres[fret].endString + difference;
+            
+            for (auto& b: m_barres[fret]) {
+                if (b.startString + difference <= 0) {
+                    int startString = std::max(0, b.startString + difference);
+                    int endString   = b.endString == -1 ? -1 : b.endString + difference;
+                    FretItem::Barre newBarre(startString, endString);
+                    m_barres[fret].push_back(newBarre);
+                }
+            }    
         }
```

Comment:
The logic here seems not entirely correct yet. So this method is about what should happen when the number of strings changes.

The old logic seems to be:
- if the number of strings changes in such way that the start string falls off the diagram, remove the entire barre
- otherwise, keep the barre, but adjust the start and end string. 

The new logic seems to do not quite that...